src/ingestors/transient-base/index.ts

```diff
@@ -0,0 +1,114 @@
+import { TRANSIENT_BASE_ABI } from './abi';
+
+export class TransientIngestor implements MintIngestor {
+  configuration = {
+    supportsContractIsExpensive: true,
+  };
+
+  async supportsUrl(resources: MintIngestorResources, url: string): Promise<boolean> {
+    if (new URL(url).hostname !== 'www.transient.xyz') {
+      return false;
+    }
+    try {
+      const { chainId, contractAddress } = await getTransientBaseMintByURL(resources, url);
+      return !!chainId && !!contractAddress;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+    const { chainId, contractAddress } = contract;
+    if (!chainId || !contractAddress) {
+      return false;
+    }
+    return await transientSupports(contract, resources.fetcher);
+  }
+
+  async createMintTemplateForUrl(resources: MintIngestorResources, url: string): Promise<MintTemplate> {
+    const isCompatible = await this.supportsUrl(resources, url);
+    if (!isCompatible) {
+      throw new MintIngestorError(MintIngestionErrorName.IncompatibleUrl, 'Incompatible URL');
+    }
+
+    const { chainId, contractAddress } = await getTransientBaseMintByURL(resources, url);
+
+    if (!chainId || !contractAddress) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+    return this.createMintForContract(resources, { chainId, contractAddress, url });
+  }
+
+  async createMintForContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<MintTemplate> {
+    const { chainId, contractAddress } = contract;
+    if (!chainId || !contractAddress) {
+      throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+    }
+
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('Transient');
+
+    if (contract.url) {
+      mintBuilder.setMarketingUrl(contract.url);
+    }
+
+    // asides name and image no other metadata we need is onchain so we can just use the offchain metadata
+    const {
+      name,
+      image,
+      description,
+      priceInWei,
+      mintAddress,
+      public_sale_start_at,
+      public_sale_end_at,
+      token_id,
+      user,
+    } = await getTransientBaseMintByAddressAndChain(resources, contract.chainId, contract.contractAddress);
+
+    mintBuilder.setName(name).setDescription(description).setFeaturedImageUrl(image);
+
+    mintBuilder.setMintOutputContract({
+      chainId,
+      address: contractAddress,
+    });
+
+    mintBuilder.setCreator({
+      name: user.name,
+      imageUrl: user.image,
+      websiteUrl: user.website,
+    });
+
+    mintBuilder.setMintInstructions({
+      chainId,
+      contractAddress: mintAddress,
+      contractMethod: 'purchase',
+      contractParams: `[address, ${token_id}, address, 1, 0, []]`,
```

Comment:
If you merge the latest from mobile-minting main, it will now run a simulation for you in running the tests.

This reveals that these params are incorrect.

Consulting the ABI:

```
{ internalType: 'address', name: 'nftAddress', type: 'address' },
      { internalType: 'uint256', name: 'tokenId', type: 'uint256' },
      { internalType: 'address', name: 'recipient', type: 'address' },
      { internalType: 'uint256', name: 'numberToMint', type: 'uint256' },
      { internalType: 'uint256', name: 'presaleNumberCanMint', type: 'uint256' },
      { internalType: 'bytes32[]', name: 'proof', type: 'bytes32[]' },
```
The important one is...

Param 1: This is the Address of the NFT collection, not the user

Replacing this with:
```
contractParams: `["${contractAddress}", "${token_id}", address, 1, 0, []]`,
```
fixes one part of the problem

The next is that the price wei for the test is being set to `20000000000000000`, but minting on metamask reveals it's actualy 0.00209.

![CleanShot 2024-07-20 at 15 20 15@2x](https://github.com/user-attachments/assets/24921ad9-4948-4a6e-a728-2bcca7a96a2d)


The priceWei is missing the mint fee it seems. 

Hopefully this helps get it across the line!

Reply:
Thanks @chrismaddern !
I figured the fee some hours ago

However, I noticed there are about 4 contract types on Base Network that Transient supports today
- ERC721TL
- ERC1155TL
- ER7160TL (The contract for Kansas smile)
- Trace

I'll be adding support for them before the next review

Reply:
All tests are passing now tho
Pls let me know if I should proceed with adding support for the remaining 3 contract types

Reply:
Amazing, yes — support for all Base supported types would be extremely helpful. Particularly the 721TL is important (and I think most of their mints?)